webhooks/services.py

```diff
@@ -1,14 +1,113 @@
 """Services module."""
 
+import json
+from json.decoder import JSONDecodeError
+from typing import List, Union
+import re
+
 from aioredis import Redis
 
+from models import (
+    BasicMessagesHook,
+    TopicItem,
+    to_credentential_hook_model,
+    to_proof_hook_model,
+    to_connections_model,
+)
+
 
 class Service:
     def __init__(self, redis: Redis) -> None:
         self._redis = redis
 
-    async def add_topic_entry(self, topic: str, hook: str):
+    def _is_proof(self, topic: str) -> bool:
+        proofs_identifiers = ["present_proof", "present_proof_v2_0"]
+        return topic in proofs_identifiers
+
+    def _is_credential(self, topic: str) -> bool:
+        credential_identifiers = ["issue_credential", "issue_credential_v2_0"]
+        return topic in credential_identifiers
+
+    def _proof_hook_versioned(self, item: dict) -> dict:
+        return to_proof_hook_model(item=item)
+
+    def _credential_hook_versioned(self, item: dict) -> dict:
+        return to_credentential_hook_model(item=item)
+
+    def _version_connections(self, item: dict) -> dict:
+        return to_connections_model(item=item)
+
+    def _deserialise(self, data: bytes) -> Union[str, dict]:
+        # Turn bytes from redis into str
+        try:
+            return (
+                data.decode("utf-8")
+                .replace("'", '"')
+                .replace("False", json.dumps(False))
+                .replace("True", json.dumps(True))
+            )
```

Comment:
Why is this needed? We do a json.dumps before sending the data to redis right? Couldn't we do a json.loads here? This feels quite hacky

Reply:
json.loads unfortunately doesn't cut the mustard here. That's what I tried first to no avail.

Reply:
You receive a byte stream from redis that you need to convert and the webhook itself is not always json conform so if we don't do this we get JSONDecode errors.